### PR TITLE
Enrich ambient architecture page

### PR DESCRIPTION
**Pull request by pegasas**
> Consider enrich ambient [architecture page](https://preliminary.istio.io/latest/docs/ops/ambient/architecture/) with some contents from the [ambient announcement blog](https://istio.io/latest/blog/2022/introducing-ambient-mesh/):
 
I mainly enrich the Differences from the sidecar architecture part,
I think this is the major architecture thinking of ambient mesh.

Also, I directly link the image to the blog to keep alignment.

Feel free to comment or modify directly if any concerns.

- [x] Ambient
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by pegasas** [comment]
try to fix https://github.com/istio/istio.io/issues/13480

**Review by Arhell — COMMENTED** [review]
/ok-to-test

**Comment by pegasas** [comment]
Hi, @Arhell , @linsun, 
Would you like to review my PR anytime you are available?
Feel free to comment or edit if any concerns.
Thanks!

File: content/en/docs/ops/ambient/architecture/index.md [hunk 1]
```diff
@@ -6,10 +6,72 @@ owner: istio/wg-networking-maintainers
...
+Traditionally, Istio implements all data plane functionality, from basic encryption through advanced L7 policy, in a single architectural component: the sidecar.
+In practice, this makes sidecars an all-or-nothing proposition.
+Even if a workload just needs simple transport security, administrators still need to pay the operational cost of deploying and maintaining a sidecar.
+Sidecars have a fixed operational cost per workload that does not scale to fit the complexity of the use case.
+
+Ambient mesh takes a different approach.
+It splits Istio’s functionality into two distinct layers.
+At the base, there’s a secure overlay that handles routing and zero trust security for traffic.
+Above that, when needed, users can enable L7 processing to get access to the full range of Istio features.
+The L7 processing mode, while heavier than the secure overlay, still runs as an ambient component of the infrastructure, requiring no modifications to application pods.
```
Comment (dhawton) [inline content/en/docs/ops/ambient/architecture/index.md:39]:
```suggestion
The L7 processing mode, while heavier than the secure overlay, runs as a component of the infrastructure, requiring no modifications to application pods.
```

File: content/en/docs/ops/ambient/architecture/index.md [hunk 2]
```diff
@@ -6,10 +6,72 @@ owner: istio/wg-networking-maintainers
...
+Importantly, from a Kubernetes perspective, waypoint proxies are just regular pods that can be auto-scaled like any other Kubernetes deployment.
+We expect this to yield significant resource savings for users, as the waypoint proxies can be auto-scaled to fit the real time traffic demand of the namespaces they serve, not the maximum worst-case load operators expect.
+
+{{< image width="100%"
+    link="/blog/2022/introducing-ambient-mesh/ambient-waypoint.png"
+    caption="When additional features are needed, ambient mesh deploys waypoint proxies, which ztunnels connect through for policy enforcement"
+    >}}
+
+Ambient mesh uses HTTP CONNECT over mTLS to implement its secure tunnels and insert waypoint proxies in the path, a pattern we call HBONE (HTTP-Based Overlay Network Environment). HBONE provides for a cleaner encapsulation of traffic than TLS on its own while enabling interoperability with common load-balancer infrastructure. FIPS builds are used by default to meet compliance needs. More details on HBONE, its standards-based approach, and plans for UDP and other non-TCP protocols will be provided in a future blog.
```
Comment (dhawton) [inline content/en/docs/ops/ambient/architecture/index.md:71]:
```suggestion
Ambient mesh uses HTTP CONNECT over mTLS to implement its secure tunnels and insert waypoint proxies in the path, a pattern we call HBONE (HTTP-Based Overlay Network Environment). HBONE provides for a cleaner encapsulation of traffic than TLS on its own while enabling interoperability with common load-balancer infrastructure.
```

**Review by dhawton — COMMENTED** [review]
Couple change recommendations. Overall this reads more like a blog than documentation.

**Review by craigbox — COMMENTED** [review]
I'm actually fine with it reading more like a blog.  We need to have overview pages in the docs which explain everything and I'm going to suggest we pull a bunch of content out of the blog to achieve this 🙂

**Comment by pegasas** [comment]
> I'm actually fine with it reading more like a blog. We need to have overview pages in the docs which explain everything and I'm going to suggest we pull a bunch of content out of the blog to achieve this 🙂

Sure. 
I agree with that point.

**Pull request closed without merging**